src/visualizers/VisualizerCycles.ts

```diff
@@ -0,0 +1,270 @@
+    }
+    while (m !== 1n) {
+        // if m is even, take the floor of it's square root
+        if (m % 2n == 0n) {
+            m = floorSqrt(m)
+            steps++
+        } else if (m % 2n !== 0n) {
+            // WARNING The function will fail after too many iterations
+            // if m is odd, take the floor of it's square root cubed
+            m = floorSqrt(m ** 3n)
+            steps++
+        }
+    }
+    // casting 'steps' to a number will not create
+    // issues as 'steps' will always be very small
+    return Number(steps)
+}
+
+class VisualizerCycles extends VisualizerDefault {
+    name = 'Cycles'
+    numberOfTerms = 1000
+    width = 50
+    cycleFormula = cycleFormula.Collatz
+    displayBoxLines = false
+    workingSequence = [] as number[]
+    params = {
+        numberOfTerms: {
+            value: this.numberOfTerms,
+            forceType: 'integer',
+            displayName: 'Number of Terms',
+            required: true,
+        },
+        width: {
+            value: this.width,
+            forceType: 'integer',
+            displayName: 'Mod of Array (width)',
+            required: true,
+        },
+        cycleFormula: {
+            value: this.cycleFormula,
+            from: cycleFormula,
+            displayName: 'Cycle Formula',
+            required: true,
+        },
+        displayBoxLines: {
+            value: this.displayBoxLines,
+            displayName: 'Overlay Box Lines',
+            required: false,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+        if (this.params.numberOfTerms.value <= 0) {
+            status.isValid = false
+            status.errors.push('Number of terms must be positive')
+        }
+        if (this.params.width.value <= 0) {
+            status.isValid = false
+            status.errors.push('Width must be positive')
+        }
+        return status
+    }
+
+    getlength(n: bigint): number {
+        return n.toString().length
+    }
+
+    setup() {
+        console.log('Success')
+
+        //assorted settings
+        this.sketch.colorMode(this.sketch.HSB)
+        this.sketch.background(0, 0, 0)
+        this.sketch.textSize(13)
+
+        // Convert the input sequence to an array
+        this.workingSequence = [] as number[]
+        const end = Math.min(
+            this.seq.first + this.numberOfTerms - 1,
+            this.seq.last
+        )
+
+        // maxCycle will be the largest number of steps that
+        // any of the starting values takes to reach 1
+        let maxCycle = 1
+
+        // The user selects either collatz or juggler collatz
+        const stepFinder =
+            this.cycleFormula === cycleFormula.Collatz
+                ? collatzStepsTo1
+                : jugglerStepsTo1
+
+        for (let i = this.seq.first; i <= end; i++) {
+            this.workingSequence.push(stepFinder(this.seq.getElement(i)))
+        }
+        //Find the maxCycle under collatz
```

Comment:
Here's one for @liam and lint:  some of the comments have a space between ```//``` and the text, and others don't.  Does anyone care? :) (I actually don't, but I'm getting the impression that things like whitespace are important sometimes...)

Reply:
Clyde appears to have made this completely uniform, so I am taking the liberty of resolving this conversation.